core/events.md

```diff
@@ -0,0 +1,520 @@
+
+This metadata (`data`) is NOT REQUIRED to be present in the event, but is
+RECOMMENDED.
+
+## Entity Events
+
+This section defines which `<ACTION>` values are applicable for each `<ENTITY>`
+value.
+
+### `registry` Events
+
+- Action: `updated`
+  - MUST be generated when a Registry's attribute is updated.
+  - MUST be generated when a Group is created or deleted, where `changed`
+    includes (at least) `epoch`, `modifiedat`, `<GROUPS>` and `<GROUPS>count`
+    attributes names.
+    TODO: GROUPS*??
+  - MUST be generated when `model`, `modelsource` or `capabilities` are
+    updated.
+
+The `created` or `updated` events are not defined as part of this specification
+as those operations are not defined in the xRegistry specification. It is
+expected that the tooling that performs those operations will define those
+events, if needed.
+
+### `model` Events
+
+- Action: `updated`
+  - TBD.
+
+### `modelsource` Events
+
+- Action: `updated`
+  - TBD.
+
+### `capabilities` Events
+
+- Action: `updated`
+  - TBD.
+
+### `group` Events
+
+- Action: `created`
+  - MUST be generated when a new Group is created.
+
+- Action: `updated`
+  - MUST be generated when a Group's attribute is updated.
+  - MUST be generated when a Resource is created or deleted, where `changed`
+    includes (at least) `epoch`, `modifiedat`, `<RESOURCES>` and
+    `<RESOURCES>count` attributes names.
+    TODO: RESOURCES*??
+
+- Action: `deleted`
+  - MUST be generated when a Group is deleted.
+
+### `resource` Events
+
+- Action: `created`
+  - MUST be generated when a new Resource is created.
+  - Note that at least one `io.xregistry.version.created` event will also be
+    being generated since at least one Version will also be created.
+
+- Action: `updated`
+  - MUST be generated when a Resource's (default Version) attribute is updated.
+  - MUST be generated when a Resource's `meta` attribute is updated, where
+    `changed` includes (at least) the changed top-level `meta` attributes
+    names prefixed with `meta.`. For example, `meta.defaultversionid`.
+    Note, this includes any updates to the `deprecated` sub-object, even
+    though a `deprecation` event is also generated.
+    TODO: meta.deprecated.* too??
+  - MUST be generated when a Version is created or deleted, where `changed`
+    includes (at least) `meta.epoch`, `meta.modifiedat`, `<VERSIONS>` and
+    `<VERSIONS>count` attributes names.
+    TODO: VERSIONS*??
+
+  Note that updating a Resource's `meta.defaultversionid` attribute will
+  generate a `io.xregistry.resource.updated` event but it MUST NOT generate
+  a `io.xregistry.version.updated` event for either the prior or new default
+  Version.
+
+- Action: `deprecation`
+  - MUST be generated when a Resource's `meta.deprecated` sub-object is
+    updated, where "changed" includes the list of top-level attribute names
+    from the `deprecated` sub-object that were modified.
+
```

Comment:
```suggestion

Note that the same event will be generated when the `deprecation` sub-object is set, any of its attributes updated, or removed (set to NULL).
```

Reply:
isn't this a repeat of the previous sentence?

Reply:
I guess I wanted to be very explicit. 🫣